### DISCUSSION TITLE
Kik Camera Tag

### DISCUSSION THREAD

**Discussion by karate21** [tomer8007/kik-bot-api-unofficial#185] [Q&A]
Hello!

Do you know of any way to override the Kik camera tag using this repo? Can the data be edited so that it looks as if a photo was sent from a camera?

Thank you!

**Comment by tomer8007** [comment]
Hi, 
I think what you request is already discussed in #130.

**Reply by karate21** [reply]
What is the <k tag that @bluemods is referring to?

**Reply by tomer8007** [reply]
You can see this tag getting built [here](https://github.com/tomer8007/kik-bot-api-unofficial/blob/d64551195c5afb5b03c854a5e7f84a77c98fe06c/kik_unofficial/utilities/cryptographic_utilities.py#L124).

**Reply by karate21** [reply]
Got it. Thank you! I'm still not quite understanding how the auth/anon stanza allow a bot to send camera images. Could you please explain this?

**Reply by bluemods** [reply]
In 2016, modded Kiks were becoming very popular, the main feature back then was to send fake camera messages. Kik did not like this, so they implemented a patch to check the xml to make sure it matched exactly what a real client should, and if it fails, it will change camera messages to gallery messages. It was dubbed the "fake camera patch". It quickly backfired in my opinion, because it let people know that modified applications were detectable. The check was defeated soon after, and Kik lost a way to stealthily detect modified applications.

**Reply by karate21** [reply]
Thank you! Do you know the correct way to code the xml so that it can't be detected?
